plugins/modules/azure_rm_recoveryservicesvault_config.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2025 Klaas Demter (@Klaas-)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: azure_rm_recoveryservicesvault_config
+version_added: "3.4.0"
+short_description: Set Azure Backup Vault Config
+description:
+    - Get Azure Backup Vault Config
+    - Most of the information here is also part of azure_rm_recoveryservicesvault
+    - it's seperated out because of https://github.com/Azure/azure-rest-api-specs/issues/34218
+    - A vault config has to exist, so you can not change the state, you can only change values, only if they are overwriteable
+    - Supports diff mode
+    - Check mode assumes all changes you make are legal and accepted by the API
+
+options:
+    vault_name:
+        description:
+            - The name of the Recovery Services Vault the vault config belongs to.
+        required: true
+        type: str
+    resource_group:
+        description:
+            - The name of the resource group the vault is in.
+        required: true
+        type: str
+    properties:
+        description:
+            - The recovery service vault config properties.
+            - https://learn.microsoft.com/en-us/python/api/azure-mgmt-recoveryservicesbackup/
+              azure.mgmt.recoveryservicesbackup.activestamp.models.backupresourcevaultconfig?view=azure-python
+        type: dict
+        suboptions:
+            enhanced_security_state:
+                description:
+                    - Soft delete and security settings for hybrid workloads
+                    - Enables soft delete, MFA and alert notifications for workloads running on premises.
+                    - Refer to https://learn.microsoft.com/en-us/azure/backup/backup-azure-security-feature#minimum-version-requirements
+                    - for minimum version requirements
+                    - Can be reverted, unless set to AlwaysON
+                type: str
+                choices:
+                    - Disabled
+                    - Enabled
+                    - AlwaysON
```

Comment:
```suggestion
                    - AlwaysON
                    - Invalid
```

Reply:
Invalid is not a value you can set via API, even if docs say you can, I have complained to PG about this